fenix/app/src/main/java/org/mozilla/fenix/components/metrics/MetricsUtils.kt

```diff
@@ -100,46 +104,49 @@ object MetricsUtils {
      * We want users using more than one of our products to report a different
      * ID in each of them. This function runs off the main thread and is CPU-bound.
      *
+     * The `customSalt` parameter allows for dynamic setting of the salt for
+     * certain experiments or any one-off applications.
+     *
      * @return an hashed and salted Google Advertising ID or null if it was not possible
      *         to get the Google Advertising ID.
      */
-    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    suspend fun getHashedIdentifier(context: Context): String? = withContext(Dispatchers.Default) {
-        getAdvertisingID(context)?.let { unhashedID ->
-            // Add some salt to the ID, before hashing. For this specific use-case, it's ok
-            // to use the same salt value for all the hashes. We want hashes to be stable
-            // within a single product, but we don't want hashes to be the same across different
-            // products (e.g. Fennec vs Fenix).
-            val salt = getHashingSalt()
+    suspend fun getHashedIdentifier(context: Context, customSalt: String? = null): String? =
+        withContext(Dispatchers.Default) {
+            getAdvertisingID(context)?.let { unhashedID ->
+                // Add some salt to the ID, before hashing. For this specific use-case, it's ok
+                // to use the same salt value for all the hashes. We want hashes to be stable
+                // within a single product, but we don't want hashes to be the same across different
+                // products (e.g. Fennec vs Fenix).
+                val salt = customSalt ?: getHashingSalt()
```

Comment:
Will we ever use the same salt for two different pings?

I don't think we'll probably touch this again, but as I type this I think to be kind to our future selves we may want to either touch up the comment above or be more explicit about the usage of this function by always passing in the salt and moving the `getHashingSalt()` to the other ping that uses this.

Reply:
Currently every other instance where we has uses the same salt. I added customSalt just to add some flexibility in case we ever need to do this sort of thing again, which with certain experiments, could be possible? It was already mentioned in the future if we ever run this experiment again, they will likely want to change the salt to better identify which experiment the ping came from.

That is a good point. I will update that comment to be more clear as to what is happening. If you are think that making the function explicitly take the salt as a parameter, I can do that too.

Reply:
I updated this in https://github.com/mozilla-mobile/firefox-android/pull/861/commits/5a77fc991493935ba9b876ff01794570d3a4c26a, let me know if this makes things clearer or if there is a specific way you'd like this worded.

I can also update the function too to just require the salt as a param.

Reply:
Looks good to me :)